src/cache/cache_las.c

```diff
@@ -461,6 +461,33 @@ __wt_las_page_skip(WT_SESSION_IMPL *session, WT_REF *ref)
 	return (skip);
 }
 
+/*
+ * __wt_las_page_obsolete_check --
+ *	 Discard references from cache to obsolete lookaside history.
+ */
+void
+__wt_las_page_obsolete_check(WT_SESSION_IMPL *session, WT_REF *ref)
+{
+	uint32_t previous_state;
+
+	if ((previous_state = ref->state) != WT_REF_LIMBO &&
+	    previous_state != WT_REF_LOOKASIDE)
+		return;
+
+	if (!__wt_atomic_casv32(&ref->state, previous_state, WT_REF_LOCKED))
+		return;
+
+	if (!__wt_page_las_active(session, ref)) {
+		__wt_free(session, ref->page_las);
+		if (previous_state == WT_REF_LOOKASIDE)
+			WT_PUBLISH(ref->state,
+			    ref->addr == NULL ?  WT_REF_DELETED : WT_REF_DISK);
```

Comment:
`NULL ?  WT_REF_DELETED` has extra white-space after `?`

Reply:
Structurally, I mildly prefer moving this code into `__evict_child_check()`: (1) that's the only caller, (2) I prefer eviction "own" updating the `WT_REF` state, (3) it matches the existing `WT_REF_DELETED` code, but I don't feel that strongly about it.

Reply:
@keithbostic, it would be great to have clear layering for `ref->state` changes (e.g., to reads, truncates and eviction).  I'll move this one (unless I can get rid of it entirely), but there's more work to do...